docs(aio): update sidenav layout, remove corresponding files

kapunahelewong opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe: update sidenav layout/order
```

**What is the current behavior?** (You can also link to an open issue here)
Current order of nav items.


**What is the new behavior?**
New organization of nav items. Remove changelog, quickstart, webpack, update sidenav, upd…  …
…ate tutorial intro, rename/move appmodule


**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:
There is a placeholder nav item for Common Templating Patterns, Expressions and Statements, and Forms Overview.

mary-poppins commented:
The angular.io preview for 3208ec56929bee448626e1f3dea379c2b519dc91 is available [here][1].

[1]: https://pr16934-3208ec56929bee448626e1f3dea379c2b519dc91.ngbuilds.io/

mary-poppins commented:
The angular.io preview for cc697b20e738d3d8128bcbe31fd20bc0250da37c is available [here][1].

[1]: https://pr16934-cc697b20e738d3d8128bcbe31fd20bc0250da37c.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 9a9f27b8439336ca66b866039baac421c613897a is available [here][1].

[1]: https://pr16934-9a9f27b8439336ca66b866039baac421c613897a.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 5d5c81028c93f7f680d1e2533173013024f87039 is available [here][1].

[1]: https://pr16934-5d5c81028c93f7f680d1e2533173013024f87039.ngbuilds.io/

petebacondarwin commented on aio/content/guide/bootstrapping.md:
```diff
@@ -1,14 +1,14 @@
 
 @intro
-Tell Angular how to construct and bootstrap the app in the root "AppModule".
+Tell Angular how to construct and bootstrap the app in the root `AppModule`.
```
You should remove all `@intro` blocks now, which then leads to the fact that you can also move the `@title` block to an h1 tag and then drop `@title` and `@description` too.

kapunahelewong replied:
Thanks, @petebacondarwin! So they will change from:

`@title
Bootstrapping

@intro
Tell Angular how to...

@description
`

to just: 

`# Bootstrapping`

?

petebacondarwin replied:
Right! :-)

mary-poppins commented:
The angular.io preview for b3953c42b5f335d0f2f8b86b121b03329f10cc4c is available [here][1].

[1]: https://pr16934-b3953c42b5f335d0f2f8b86b121b03329f10cc4c.ngbuilds.io/

mary-poppins commented:
The angular.io preview for e56694fbbe77d9f1495465263526860d0123dec8 is available [here][1].

[1]: https://pr16934-e56694fbbe77d9f1495465263526860d0123dec8.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 4d8f781559a9170fab56907d4b5d4d8c4bc1672f is available [here][1].

[1]: https://pr16934-4d8f781559a9170fab56907d4b5d4d8c4bc1672f.ngbuilds.io/

IgorMinar commented on aio/content/navigation.json:
```diff
@@ -350,49 +310,28 @@
-          "url": "guide/styleguide",
-          "title": "Style Guide",
-          "tooltip": "Write Angular with style."
+          "url": "guide/cheatsheet",
```
can we move fundamentals to the foundations? maybe as the last item there?

kapunahelewong replied:
@IgorMinar I'm not sure where Foundations is. Could you clarify? Thanks!

IgorMinar replied:
sorry!!! I meant, can we move the cheatsheet into the fundamentals section as the last item there.

kapunahelewong replied:
@IgorMinar Aha! Sure, thanks! :)

IgorMinar commented on aio/content/navigation.json:
```diff
@@ -27,23 +27,13 @@
     },
 
     {
+      "url": "guide/cli-quickstart",
```
can we rename the doc to be called just "quickstart" ?

kapunahelewong replied:
Done.

IgorMinar commented on aio/content/navigation.json:
```diff
@@ -148,24 +105,88 @@
+              "tooltip": "A form creates a cohesive, effective, and compelling data entry experience. An Angular form coordinates a set of data-bound user controls, tracks changes, validates input, and presents errors."
+            },
+            {
+              "url": "guide/dynamic-form",
```
this is one depends on reactive forms, can we move it just after reactive forms?

kapunahelewong replied:
Done.

IgorMinar commented on aio/content/navigation.json:
```diff
@@ -148,24 +105,88 @@
               "tooltip": "Load components dynamically."
             },
             {
+              "url": "guide/template-syntax",
```
can we move this just after displaying-data?

kapunahelewong replied:
Done.

mary-poppins commented:
The angular.io preview for 05f558e5c9e526d7fdb7b4fcfb78e3cb2c7ca568 is available [here][1].

[1]: https://pr16934-05f558e5c9e526d7fdb7b4fcfb78e3cb2c7ca568.ngbuilds.io/

mary-poppins commented:
The angular.io preview for efb356c81f1c2315e2f0e38e207caebfd53de2c8 is available [here][1].

[1]: https://pr16934-efb356c81f1c2315e2f0e38e207caebfd53de2c8.ngbuilds.io/

kapunahelewong committed:
docs(aio): reorganise the sidenav menu

Reorganizes the items in the sidenav menu and consolidates the quickstart
and cli-quickstart guides into one.

mary-poppins commented:
The angular.io preview for 6b4507aac94553c112b97ebaca750e617f81c24f is available [here][1].

[1]: https://pr16934-6b4507aac94553c112b97ebaca750e617f81c24f.ngbuilds.io/

IgorMinar commented:
Shouldn't "Getting Started" be in all caps to be consistent with the rest of the top level menu items?
<img width="261" alt="screen shot 2017-05-24 at 12 31 26 pm" src="https://cloud.githubusercontent.com/assets/216296/26421997/fbd2b99a-407c-11e7-82ed-a2c484b631c5.png">

petebacondarwin commented:
That is a CSS issue - currently only "container" nodes are given upper case style.

IgorMinar commented:
@petebacondarwin "API" is not a container element. Is it just that it happens to be all caps in the source form?

petebacondarwin commented:
Precisely

mary-poppins commented:
The angular.io preview for 21f28205211c000a2e48dcec843e828a02c4d736 is available [here][1].

[1]: https://pr16934-21f28205211c000a2e48dcec843e828a02c4d736.ngbuilds.io/

petebacondarwin commented on packages/examples/platform-browser/browser/title.ts~HEAD:
```diff
@@ -0,0 +1,26 @@
+// #docplaster
```
Is the filename correct here?

kapunahelewong replied:
@petebacondarwin Oh my goodness, thank you for spotting that.

mary-poppins commented:
The angular.io preview for 3627752a8d3d075514e3921d6d09bc462d78f256 is available [here][1].

[1]: https://pr16934-3627752a8d3d075514e3921d6d09bc462d78f256.ngbuilds.io/

IgorMinar commented:
I rebased this PR and force-pushed into this branch

petebacondarwin commented:
I have assumed that the `title` example was added by mistake but then instead of being deleted the actual `Title` service source file was deleted. I am reverting both of those from this PR, since I believe they are not relevant and we can easily reproduce later.

Also I have removed all links from sidenav container nodes, since the UI is no longer designed to support that; and the urls to which they linked are the first items of their children anyway.

Updated and force pushed

mary-poppins commented:
The angular.io preview for 7b33e11e3cd3d6078989c785aa9af18e5f96b8eb is available [here][1].

[1]: https://pr16934-7b33e11e3cd3d6078989c785aa9af18e5f96b8eb.ngbuilds.io/

mary-poppins commented:
The angular.io preview for d9c544d1d4342de23b3b316d7859baa8640e4de9 is available [here][1].

[1]: https://pr16934-d9c544d1d4342de23b3b316d7859baa8640e4de9.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 5569f6ea99e12bc029c32c59c09b51037fcde166 is available [here][1].

[1]: https://pr16934-5569f6ea99e12bc029c32c59c09b51037fcde166.ngbuilds.io/

mary-poppins commented:
The angular.io preview for e4a3c5c24b6ed04383e6bde97850fba3ee7f8646 is available [here][1].

[1]: https://pr16934-e4a3c5c24b6ed04383e6bde97850fba3ee7f8646.ngbuilds.io/

mary-poppins commented:
The angular.io preview for ee1e2e9dd78c514b12f521c02b4dc3a9b00d2c48 is available [here][1].

[1]: https://pr16934-ee1e2e9dd78c514b12f521c02b4dc3a9b00d2c48.ngbuilds.io/

mary-poppins commented:
The angular.io preview for ca696936af68b003f27f15a8a4872af12f7a19e2 is available [here][1].

[1]: https://pr16934-ca696936af68b003f27f15a8a4872af12f7a19e2.ngbuilds.io/
